### PR TITLE
Fix bug related to empty normalized tables

### PR DESCRIPTION
**Pull request by mikeknep**
When parsing JSON, it's possible for all values to contain some empty list value—for example, in the unit tests `bball` fixture, each player has a `suspensions` key with empty list value. We want to be able to restore this field in the final denormalized/original-matching output data, but we **don't** want to submit an empty table for training.

We now include add empty invented tables to the graph (so we know where to put them back when denormalizing), but ensure they are not exposed to clients asking for MODELABLE tables.

As before this branch, we still do not support dataframes where _all columns_ contain JSON lists. Fortunately, as the unit test changes demonstrate, these changes have the added benefit of detecting that sooner and avoiding ending up in a bad state when this edge case happens.

### CODE REVIEW

File: src/gretel_trainer/relational/backup.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 @dataclass
 class BackupRelationalDataTable:
     primary_key: list[str]
-    invented_table_metadata: Optional[dict[str, str]] = None
+    invented_table_metadata: Optional[dict] = None
```
Comment (pimlock) [inline src/gretel_trainer/relational/backup.py:13]: Would it be worth keeping the first `str` and putting `Any` for value types?
I think we don't use any of these values directly, they are only used to populate `InventedTableMetadata`, right?

**Review by pimlock — APPROVED** [review]

File: src/gretel_trainer/relational/backup.py (hunk 1 above)
Comment (mikeknep) [reply src/gretel_trainer/relational/backup.py]: Yeah good call. I feel like `dict` should _be_ `dict[str, Any]` instead of I guess effectively `dict[Any, Any]`.

You're correct that this dataclass is strictly capturing what goes in and out of the backup file, and the values are just used to populate the `InventedTableMetadata` object. I feel like it's useful to have a pretty strict definition of what to expect to see in the backup JSON files, but it does feel a little obnoxious at times since many structures are nearly or even exactly identical to the internal domain objects used more widely in the code.
